host/src/udf_query.rs

```diff
@@ -0,0 +1,171 @@
+//! Embedded SQL approach for executing Python UDFs within SQL queries.
```

Comment:
I wonder if we should split the host into two crates: the mere UDF wrapper (= low level, might be used by people that don't wanna use SQL or Python) and the SQL+Python stuff (which then can also depend on whatever comes out of #17).

Reply:
Actually: can we drop the word "Python" from this module (not the tests, just the actual impl.) altogether? This code technically works with all our `WasmComponentPrecompiled`s because they just define "give me some source text". That choice was deliberate because we could easily use the same interface later to bundle other guest languages like Lua or JavaScript or something.